cpg_workflows/stages/large_cohort.py

```diff
@@ -96,8 +96,8 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
 
         j: PythonJob = get_batch().new_python_job('Combiner', (self.get_job_attrs() or {}) | {'tool': HAIL_QUERY})
         j.image(image_path('cpg_workflows'))
-        j.memory(combiner_config.get('memory', "4Gi"))
-        j.storage(combiner_config.get('storage', "5Gi"))
+        j.memory(combiner_config.get('memory', '4Gi'))
+        j.storage(combiner_config.get('storage', '5Gi'))
```

Comment:
I think it's confusing to have both defaults - if you set 8Gi/10Gi in the large_cohort.toml, then this 4Gi/5Gi can be removed (it's always overwritten anyway, and it will just lead to confusion about where 8Gi is coming from when 4Gi is apparently the default). Or, set the default here so it's easy to find in the code, but don't overwrite it in the toml.

It doesn't look like we have a consistent approach through the large cohorts pipeline, not sure if others (@MattWellie?) have been more consistent elsewhere.

Reply:
I've been crap at this in the past, sometimes leaving the in-code default slightly different to the in-config default as things evolve. I'm trying to fix all those instances, and I'd generally recommend the code and config defaults be identical.

I think there's utility in this in-code default being there so that you have the at-a-glance estimate of how big (expensive) each job is just from reading the code, without having the config open alongside

Reply:
I personally like the in-code defaults to get a quick estimate of job size without looking into the config, but it can muddle things. A comment would be equally clear, and would make the pipeline fail if a default isn't set in config (for pipelines it always is). The pipeline is woefully short of code conventions to follow, maybe we should discuss some of those syntax (in)consistencies when we come to fragmenting the codebase.

There's a tangent to go down about scripts which can run in the pipeline context or standalone, but that's not relevant to any pipeline stage code.

Reply:
I personally like the in-code defaults to get a quick estimate of job size without looking into the config, but it can muddle things. A comment would be equally clear, and would make the pipeline fail if a default isn't set in config (for pipelines it always is). The pipeline is woefully short of code conventions to follow, maybe we should discuss some of those syntax (in)consistencies when we come to fragmenting the codebase.

There's a tangent to go down about scripts which can run in the pipeline context or standalone, but that's not relevant to any pipeline stage code.

Reply:
These have been removed.

Reply:
+1 for discussing some of these conventions as part of the codebase fragmentation.

Reply:
Edited to use the configuration defaults, as per the force_new_combiner decision below (the latest revision had removed the setting of storage and memory in the combiner entirely).